cirq-core/cirq/devices/fidelity.py

```diff
@@ -0,0 +1,216 @@
+from typing import Sequence
+from itertools import product
+import cirq
+import numpy as np
+from cirq.devices.noise_model import validate_all_measurements
+
+
+class Fidelity:
+    def __init__(
+        self,
+        *,
+        t1: float = None,
+        decay_constant: float = None,
+        xeb_fidelity: float = None,
+        pauli_error: float = None,
+        p00: float = None,
+        p11: float = None,
+    ) -> None:
+        """Creates a Fidelity object using the provided metrics
+
+        Args:
+          t1: t1 decay constant in ns
+          decay_constant: depolarization decay constant
+          xeb_fidelity: 2-qubit XEB Fidelity
+          pauli_error: total Pauli error
+          p00: probability of qubit initialized as zero being measured as zero
+          p11: probability of qubit initialized as one being measured as one
+
+          Only one of decay_constant, xeb_fidelity, and pauli_error should be specified
+
+        Returns:
+          a Fidelity object with the provided metrics that can be used to create a noise model
+        """
+        if not any([t1, decay_constant, xeb_fidelity, pauli_error, p00, p11]):
+            raise ValueError('At least one metric must be specified')
+
+        for metric in [xeb_fidelity, pauli_error, p00, p11]:
+            if metric is not None and not 0.0 <= metric <= 1.0:
+                raise ValueError('xeb, pauli error, p00, and p11 must be between 0 and 1')
+
+        if (
+            np.count_nonzero(
+                [metric is not None for metric in [xeb_fidelity, pauli_error, decay_constant]]
+            )
+            > 1
+        ):
+            raise ValueError(
+                'Only one of xeb fidelity, pauli error, or decay constant should be defined'
+            )
+
+        self._t1 = t1
+        self._p = decay_constant
+        self._p00 = p00
+        self._p11 = p11
+        self._pauli_error = pauli_error
+        self._xeb = xeb_fidelity
+
+        if self._pauli_error is not None:
+            self._p = self.pauli_error_to_decay_constant()
+            self._xeb = self.pauli_error_to_xeb_fidelity()
+        elif self._p is not None:
+            self._pauli_error = self.decay_constant_to_pauli_error()
+            self._xeb = self.decay_constant_to_xeb_fidelity()
+        elif self._xeb is not None:
+            self._p = self.xeb_fidelity_to_decay_constant()
+            self._pauli_error = self.decay_constant_to_pauli_error()
```

Comment:
I would recommend storing only one of (decay_constant, pauli_error, or xeb fidelity), then using your conversion functions in the implementation of the properties below. I.e. only store self._p and use conversions for the pauli_error and xeb properties.